Day 1/DayOnePartTwo.py

```diff
@@ -0,0 +1,31 @@
+import sys
+sys.path.insert(0, "C:/Users/Callipolis/Desktop/Programming/AoC2/Advent-of-Code-2015")
+from InputReader import InputReader
+
+input_reader = InputReader("input1.txt")
+raw_input = input_reader.get_raw_input()
+
+from DayOnePartOne import FloorCounter
+
```

Comment:
Na az inheritance. Atbeszelhetjuk majd discordon is ha gondolod de leirom a lenyeget. 
Ezzel vigyazz mert egy konnyu modja a kod reuse-nak, de 10bol 9szer nem ez a jo megkozelites es akkor nagyon sok gondor tud okozni.
A key hogy akkor van inheritance ha megvan az "is a" relacio, koncepcionalisan es funkcionalisan. "A cat is an animal". 
Itt te azt mondod hogy "A BasementFinder is a FloorCounter", ami already kicsit weird, de funkcionalitasban nagyon kijon hogy nem.
A legfontosabb contract amit be kell tartanod inheritance-nel hogy az oroklott (child) object barhol hasznalhato kell legyen ahol egy parent object hasznalhato. Jelen esetben a part one-t meg kene tudnod oldani egy basement finder-rel. Ez nem fog menni mert a count floors leall a basement-nel, tehat nem tudja azt amit a parent.

Ha ragaszkodsz az orokleshez akkor azt csinalnam hogy nem a count floors irom ujra hanem hozzaadnek egy uj fuggvenyr ami "find basement" vagy hasonlo es abba irnam a kodot.De ez altal sem lesz igazabol kozos reszhalmaza a funkcionalitasnak. A basement finder olyan osztaly akinek kell egy floor counter hogy elvegezze a dolgat de o nem egy floor counter.

Fix: En "fel"mozgatnam az is-basement fuggvenyt a floor counterre es nem orokitenem a basement findert hanem siman egy uj osztalyt irnek akinek van egy `self.floor_counter = FloorCounter(...)`-e az initben. Van egy find_basement fuggvenye aki azt csinalja mint most a count floors. Meg mondjuk egy get_basement_position vagy valami hogy a megoldast kivedd.

Reply:
Az az infó hiányzott, hogy childnak meg kell tartani az ős képességeit. BasementFinder most is örököl, de szétbontottam a methodokat és nincs semmi felülírva az ősből, csak átvesz belőle.

Reply:
https://en.wikipedia.org/wiki/Liskov_substitution_principle